cowprotocol/cowswap
# fix(token-selector): show direct match on top

# Summary

User reported issue on discord: GNO doesn't show on token selector when searching for it in the dropdown.

Also reported by Chen https://github.com/cowprotocol/token-lists/issues/313

This PR addresses it in ~3~ 4 ways:
- Direct token symbol matches are displayed on top
- Active list search results no longer capped at 10 results
- GNO added to the default favourite tokens list on Gchain
- Filter out during search the `on/from gnosis/xdai/ethereum/mainnet` from token names

# To Test

1. Load the app on Gnosic chain
2. Open the token selector
* GNO should be one default favourite token
![image](https://github.com/cowprotocol/cowswap/assets/43217/d336918e-c9aa-4fd3-aa07-aa00e1e64621)

3. Search for `gno`
* GNO token should show up on top
![image](https://github.com/cowprotocol/cowswap/assets/43217/bd6843e8-7fb4-4bc9-b384-d79839359b29)

4. Search for `gnosis`
* The list of results should not be capped at 10

~Note: `gnosis` on `gnosis chain` is a special case as the majority of tokens have `gnosis` in their name.~
~Thus it's hard for the generic filter to give `GNO` token a high position in the search.~
Updated

* The list should show only gnosis matches 
![image](https://github.com/cowprotocol/cowswap/assets/43217/e79f1acc-a9b0-491d-aaf4-b4aa1633bfed)
